homeassistant/components/nuki/config_flow.py

```diff
@@ -110,7 +114,9 @@ async def async_step_reauth_confirm(self, user_input=None):
             errors["base"] = "unknown"
 
         if not errors:
-            existing_entry = await self.async_set_unique_id(info["ids"]["hardwareId"])
+            existing_entry = await self.async_set_unique_id(
```

Comment:
Can we migrate existing config entries with the old id to the new id? Eg by checking the unique_id during config entry setup of it matches the old pattern and then apply the new id via `hass.config_entries.async_update_entry`.

Reply:
would this go into `async def async_setup_entry(hass, entry):` ?

Reply:
Yes, exactly, in the `__init__.py` module.